src/components/actions/scheduler.ts

```diff
@@ -63,14 +63,24 @@ export const cleanUpFiles = async () => {
     'Cleaning up filess and file records...'
   )
   await crawlFileSystem(basePath, checkFile)
+  processMissingFileLinks()
   const deleteCount = await DBConnect.cleanUpFiles()
   if (deleteCount > 0) console.log(`${deleteCount} files removed.`)
-  else console.log('no files removed.')
 }
 
 // Manually launch cleanup with command
 if (isManualCleanup) {
   cleanUpFiles()
 }
 
+const processMissingFileLinks = async () => {
+  const filePaths: any = await DBConnect.filePaths()
+  filePaths.rows.forEach((filePathObject: { file_path: string }) => {
```

Comment:
Also --- careful with `forEach` within `async` functions, remember? (You can use them, just need to know what you're doing and why)

Reply:
I've committed everything bar the cursor. I'm just having a read up on them now.

Reply:
Brave man! 😆

Reply:
I think I am probably missing some conceptual understanding here. 

I'm looking at the initial filePaths query - and this one doesn't have any WHERE filtering. Have you envisioned a WHERE filter here (which would be somewhat arbitrary?) to limit the size of queries to something manageable?

Reply:
No, the idea is the CURSOR would iterate over the query results one by one without loading the massive result list into memory. I can have a look at this with you tomorrow if there's time. It might not end up being worth it, I just thought it'd be worth a look.